packages/engine-server/src/drivers/file/storev2.ts

```diff
@@ -401,13 +416,22 @@ export class FileStorage implements DStore {
         };
         const vpath = vault2Path({ vault, wsRoot: this.wsRoot });
         // OPT:make async and don't wait for return
-        if (!this.engine.config.noCaching) {
+        // Skip this if we found no notes, which means vault did not initialize
+        if (!this.engine.config.noCaching && notes.length > 0) {
```

Comment:
the comment doesn't seem to reflect the if statement? `notes.length > 0`

Reply:
We only write notes to cache if there are notes to write, `notes.length > 0`. If `notes.length === 0`, then this block is skipped. Am I misreading the code?

Reply:
No, I'm being dumb 😀, please ignore.